src/ui/QGCToolBar.cc

```diff
@@ -676,10 +676,8 @@ void QGCToolBar::_connectButtonClicked(bool checked)
             // Get the configuration name
             QString confName = _linkCombo->currentText();
             // Create a link for it
-            LinkInterface* link = _linkMgr->createLink(confName);
+            LinkInterface* link = _linkMgr->createConnectedLink(confName);
             if(link) {
-                // Connect it
-                _linkMgr->connectLink(link);
                 // Save last used connection
```

Comment:
This class (old Tool Bar) is not being used. I left the code in for the time being while making sure the new tool bar was fully functional. I will be removing it soon.